src/components/step-flow/step-flow.component.tsx

```diff
@@ -0,0 +1,193 @@
+   * and allow navigation from the StepFlow component to the contents below.
+   */
+  titleRef: React.RefObject<HTMLDivElement>;
+  /** The tabindex which will be applied to the DOM element which contains the component's title.
+   * This ensures the element becomes tabbable.
+   */
+  titleTabIndex?: -1 | 0;
+  /** The total steps in the user journey.  */
+  totalSteps: Steps;
+  /**
+   * The current step of the user journey. If the set `currentStep` is higher than
+   * `totalSteps`the value of `currentStep` will be that of `totalSteps` instead.
+   */
+  currentStep: Steps;
+  /** Determines if the progress indicator is shown. */
+  showProgressIndicator?: boolean;
+  /** Determines if the close icon button is shown */
+  showCloseIcon?: boolean;
+  /** function runs when user click dismiss button */
+  onDismiss?: (
+    e:
+      | React.KeyboardEvent<HTMLButtonElement>
+      | React.MouseEvent<HTMLButtonElement>
+  ) => void;
+}
+
+let currentStepWarnTriggered = false;
+
+export const StepFlow = ({
+  category,
+  categoryTypographyVariant,
+  title,
+  titleTypographyVariant,
+  titleRef,
+  titleTabIndex = -1,
+  totalSteps,
+  currentStep,
+  showProgressIndicator = false,
+  showCloseIcon = false,
+  onDismiss,
+  ...rest
+}: StepFlowProps) => {
+  const totalStepsArray = Array.from(
+    { length: totalSteps },
+    (_, index) => index + 1
+  );
+
+  const validatedCurrentStep =
+    currentStep > totalSteps ? totalSteps : currentStep;
+
+  /* eslint-disable no-console */
+  if (!currentStepWarnTriggered && currentStep > totalSteps) {
+    currentStepWarnTriggered = true;
+    console.warn(
+      "[WARNING] The `currentStep` prop should not be higher than the `totalSteps`prop in `StepFlow`." +
+        " Please ensure `currentStep`s value does not exceed that of `totalSteps`, in the meantime" +
+        " we have set `currentStep` value to that of `totalSteps`, and all indicators have been marked as completed."
+    );
+  }
+
+  const progressIndicators = totalStepsArray.map((step) => {
+    const generateDataState = () => {
+      if (step === validatedCurrentStep) {
+        return "in-progress";
+      }
+      if (step < validatedCurrentStep) {
+        return "is-completed";
+      }
+      return "not-completed";
+    };
+
+    return (
+      <StyledProgressIndicator
+        key={step}
+        data-element="progress-indicator"
+        isCompleted={step < validatedCurrentStep}
+        isInProgress={step === validatedCurrentStep}
+        data-state={generateDataState()}
+      >
+        &nbsp;
+      </StyledProgressIndicator>
+    );
+  });
+
+  const closeIcon = (
+    <IconButton data-element="close" onClick={onDismiss}>
+      <Icon type="close" />
+    </IconButton>
+  );
+
+  const locale = useLocale();
+
+  return (
+    <StyledStepFlow {...rest} {...tagComponent("step-flow", rest)}>
+      <StyledStepContent>
+        <StyledStepContentText>
```

Comment:
Comment: this would render an empty container if no `category` is set. We could probably get away with using `Typography` directly here is we add support for `aria-hidden` etc